libs/coin-framework/src/api/types.ts

```diff
@@ -321,20 +321,47 @@ type MaybeTxData<TxDataType extends TxData> = TxDataType extends TxDataNotSuppor
 
 export type FeesStrategy = "slow" | "medium" | "fast";
 
+export type StakingOperation =
+  | "delegate"
+  | "undelegate"
+  | "redelegate"
+  | "getStakedBalance"
+  | "getUnstakedBalance";
+
 export type TransactionIntent<
   MemoType extends Memo = MemoNotSupported,
   TxDataType extends TxData = TxDataNotSupported,
 > = {
+  intentType: "transaction" | "staking";
   type: string;
```

Comment:
We now have `type` and `intentType`, I find this ambiguous :s A staking transaction is still a transaction, can't we unify ?

Reply:
So maybe `mode`  in StakingTransactionIntent and `type` express the same thing, so we could move this `type` to `mode` in `SendTransactionIntent`